strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java

```diff
@@ -325,96 +311,79 @@ public void testDeploymentRedeploymentAndDeletionAgainstGroupRepository(TestInfo
         createRepository(STORAGE0, repositoryGroup);
         // Test resource initialization end.
 
-        InputStream is = null;
-
         String gavtc = "org.carlspring.strongbox:strongbox-utils:8.3:jar";
 
         Artifact artifact = ArtifactUtils.getArtifactFromGAVTC(gavtc);
 
-        //noinspection EmptyCatchBlock
-        try
-        {
-            File repositoryDir = getRepositoryBasedir(STORAGE0, repositoryGroupId);
-            is = generateArtifactInputStream(repositoryDir.toPath().getParent().toAbsolutePath().toString(),
-                                             repositoryGroupId,
-                                             gavtc,
-                                             true);
+        File repositoryDir = getRepositoryBasedir(STORAGE0, repositoryGroupId);
 
-            RepositoryPath repositoryPath = repositoryPathResolver.resolve(STORAGE0,
-                                                                           repositoryGroupId,
-                                                                           ArtifactUtils.convertArtifactToPath(artifact));
-            mavenArtifactManagementService.validateAndStore(repositoryPath, is);
+        try (InputStream is = generateArtifactInputStream(repositoryDir.toPath().getParent().toAbsolutePath().toString(),
+                                                          repositoryGroupId,
+                                                          gavtc,
+                                                          true)){
+            //noinspection EmptyCatchBlock
+            try
+            {
+                RepositoryPath repositoryPath = repositoryPathResolver.resolve(STORAGE0,
+                                                                               repositoryGroupId,
+                                                                               ArtifactUtils.convertArtifactToPath(artifact));
+                mavenArtifactManagementService.validateAndStore(repositoryPath, is);
 
-            fail("Failed to deny artifact operation for repository with disallowed deployments.");
```

Comment:
I'm not sure I'm following the changes in the blocks below very well. Would you mind clarifying what you've changed, as it looks like you've changed/reformatted everything...?

Reply:
I introduced an outer try-with-resources inside of which are put all original try statements.
This allowed to replace multiple finally blocks with one try-with-resources.

So, the essence of the change is addition of the following code:

```
try (InputStream is = generateArtifactInputStream(repositoryDir.toPath().getParent().toAbsolutePath().toString(),
                                                  repositoryGroupId,
                                                  gavtc,
                                                  true))
{ ... }
```